server/registry/internal/storage/list.go

```diff
@@ -99,24 +119,45 @@ var deploymentFields = map[string]filtering.FieldType{
 }
 
 var artifactFields = map[string]filtering.FieldType{
-	"name":        filtering.String,
-	"project_id":  filtering.String,
-	"api_id":      filtering.String,
-	"version_id":  filtering.String,
-	"spec_id":     filtering.String,
-	"artifact_id": filtering.String,
-	"create_time": filtering.Timestamp,
-	"update_time": filtering.Timestamp,
-	"mime_type":   filtering.String,
-	"size_bytes":  filtering.Int,
+	"name":          filtering.String,
+	"project_id":    filtering.String,
+	"api_id":        filtering.String,
+	"version_id":    filtering.String,
+	"spec_id":       filtering.String,
+	"artifact_id":   filtering.String,
+	"deployment_id": filtering.String,
+	"create_time":   filtering.Timestamp,
+	"update_time":   filtering.Timestamp,
+	"mime_type":     filtering.String,
+	"size_bytes":    filtering.Int,
+}
+
+var revisionedArtifactFields = map[string]filtering.FieldType{
+	"name":                 filtering.String,
+	"project_id":           filtering.String,
+	"api_id":               filtering.String,
+	"version_id":           filtering.String,
+	"spec_id":              filtering.String,
+	"artifact_id":          filtering.String,
+	"deployment_id":        filtering.String,
+	"create_time":          filtering.Timestamp,
+	"update_time":          filtering.Timestamp,
+	"mime_type":            filtering.String,
+	"size_bytes":           filtering.Int,
+	"revision_create_time": filtering.Timestamp,
+	"revision_update_time": filtering.Timestamp,
 }
 
 // gormOrdering accepts a user-specified order_by string and returns a gorm-compatible equivalent.
 // For example, the user-specified string `name,description` returns `key,description`.
-// An error is returned if the string is invalid or refers to a field that isn't included in the provided `fields` map.
-func gormOrdering(ordering string, fields map[string]filtering.FieldType) (string, error) {
+// An error is returned if the string is invalid or refers to a field that isn't included in the `fields` map.
+func gormOrdering(ordering, table string) (string, error) {
+	fields, ok := tableFieldsLookup[table]
+	if !ok {
+		return "", status.Errorf(codes.Internal, "unknown order table: %q", table)
+	}
```

Comment:
This failure mode seems unnecessary to me. Each caller of gormOrdering should be able to provide the data at compile time, which avoids using lookup tables (and having to handle these errors) at runtime. Or am I missing something?

Reply:
You're right, good catch. A previous iteration required the table name for namespacing, but I neglected to back this out after I was able to eliminate that requirement.

Reply:
Both the fields and default order are accessed by table name, so it didn't make sense to pass in both (opening a new failure mode if they didn't match). If you'd rather, I can make the table name an enum instead of a string... I had considered that, but it just didn't seem to add a lot of value for an purely internal lookup.